src/SimulationRunner.cc

```diff
@@ -464,104 +470,60 @@ void SimulationRunner::AddSystem(
       std::optional<Entity> _entity,
       std::optional<std::shared_ptr<const sdf::Element>> _sdf)
 {
-  this->AddSystemImpl(SystemInternal(_system), _entity, _sdf);
-}
+  auto entity = _entity.has_value() ? _entity.value()
+      : worldEntity(this->entityCompMgr);
+  auto sdf = _sdf.has_value() ? _sdf.value() : this->sdfWorld->Element();
 
-//////////////////////////////////////////////////
-void SimulationRunner::AddSystemImpl(
-      SystemInternal _system,
-      std::optional<Entity> _entity,
-      std::optional<std::shared_ptr<const sdf::Element>> _sdf)
-{
-  // Call configure
-  if (_system.configure)
-  {
-    // Default to world entity and SDF
-    auto entity = _entity.has_value() ? _entity.value()
-        : worldEntity(this->entityCompMgr);
-    auto sdf = _sdf.has_value() ? _sdf.value() : this->sdfWorld->Element();
-
-    _system.configure->Configure(
-        entity, sdf,
-        this->entityCompMgr,
-        this->eventMgr);
-  }
-
-  // Update callbacks will be handled later, add to queue
-  std::lock_guard<std::mutex> lock(this->pendingSystemsMutex);
-  this->pendingSystems.push_back(_system);
+  this->systemMgr->AddSystem(_system, entity, sdf);
```

Comment:
It looks like `SystemManager::ConfigurePendingSystems` is always called after `SystemManager::AddSystem`. Does it make sense to simplify this by calling `ConfigurePendingSystems` within `AddSystem`?

Reply:
Ah, this is related to a follow on, I'll clean it up while retargeting